avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1486,11 +1486,12 @@ private Object toConnectData(Schema schema, Object value, ToConnectContext toCon
               valueRecordSchema = toConnectSchemaWithCycles(
                   valueRecord.getSchema(), true, null, null, toConnectContext);
             }
+
             for (Field field : schema.fields()) {
               Schema fieldSchema = field.schema();
-
               if (isInstanceOfAvroSchemaTypeForSimpleSchema(fieldSchema, value)
-                  || (valueRecordSchema != null && schemaEquals(valueRecordSchema, fieldSchema))) {
+                  || (valueRecordSchema != null && schemaEquals(valueRecordSchema, fieldSchema,
```

Comment:
Let's keep the method `boolean schemaEquals(Schema, Schema)` and have it simply call `return schemaEquals(s1, s2, new HashMap<>());`.   Then this line can omit the `new HashMap<>()`